### PR TITLE
Update index.d.ts in @types/ot with comments and other descriptions

### PR DESCRIPTION
**Pull request by ppoulard**
Comments + (almost) full description

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [ ] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [ ] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by danger-public** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## ot ([<kbd>unpkg</kbd>](https://unpkg.com/browse/ot@latest/))
was missing the following properties:
1. EditorSocketIOServer
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 437c31991d9aca136a03e83d1cdc7df31621f84e
</p>

**Comment by ppoulard** [comment]
Done !

File: types/ot/index.d.ts [hunk 1]
```diff
@@ -3,50 +3,477 @@
...
+    onChange(textOperation: TextOperation, inverse: TextOperation): void;
+    updateSelection(): void;
+    onSelectionChange(): void;
+    onBlur(): void;
+    sendSelection(selection: Selection): void;
+    sendOperation(revision: number, operation: TextOperation): void;
+    applyOperation(operation: TextOperation): void;
+}
+
+// TODO
```
Comment (amcasey) [inline types/ot/index.d.ts:474]: What needs to be done here?

**Review by amcasey — CHANGES_REQUESTED** [review]
Please consider adding some tests.

File: types/ot/index.d.ts (hunk 1 above)
Comment (ppoulard) [reply types/ot/index.d.ts:474]: To describe what's inside `SelfMeta` and `OtherClient`. Maybe for the next step ?

**Comment by ppoulard** [comment]
> Please consider adding some tests.

@amcasey What kind of tests would you expect ? I used those definitions to my project and they fit well. This is why I decide to send them back to the community. You may just try them yourself and you will see (feel free to fix any bug if necessary, or ask me).
Kind regards.

File: types/ot/index.d.ts (hunk 1 above)
Comment (amcasey) [reply types/ot/index.d.ts:474]: Personally, I don't like to leave bare TODOs in code, so I'd probably add an approximate description of the work that needs to be done.  However, I mostly wanted to confirm that you hadn't left it there unintentionally.

I agree that these don't need to be fleshed out in this change.

**Comment by amcasey** [comment]
@ppoulard Most types packages are accompanied by a test file that shows sample usages of the declared types.  They can either confirm that normal usage does not produce errors or use special annotations to validate that incorrect usages do produce errors.  However, I see that this package lacks tests entirely, so I don't suppose there's any urgent need to add them now.

**Review by amcasey — APPROVED** [review]

**Review by MichaelDeBoey — APPROVED** [review]

**Comment by MichaelDeBoey** [comment]
Ready to merge

**Pull request merged**